dist/bin/current/helperScripts/networkFunctions.sh

```diff
@@ -1,42 +1,59 @@
-  # Check for groovyserv. If it does not exist, try to download it. If it still fails, start Roddy without it.
+  # Check for groovyserv. If it does not exist, try to downloading it. If it still fails, start Roddy without it.
   local callerBinary=java
   local roddyApplicationDirectory=${1-""}
   local runtimeFolder=${roddyApplicationDirectory}/dist/runtime
   local forbiddenFile=${runtimeFolder}/gservforbidden
 
-  if [[ ! -d $(find ${runtimeFolder} -maxdepth 1 -mindepth 1 -type d -name "groovys*") && ! -f ${forbiddenFile} ]]; then
+  if [[ ! -d "$runtimeFolder" ]]; then
+    mkdir -p "$runtimeFolder" 2> /dev/null
+  fi
 
-    mkdir -p dist/runtime/ 2>/dev/null
-    (cd ${runtimeFolder} && wget -S $(curl -s https://kobo.github.io/groovyserv/download.html | grep groovyserv | grep bin.zip | sed 's/href=/\n/g' | sed 's/>/\n/g' | sed 's/</\n/g' | sed 's/"//g'  | grep .zip)) &> /dev/null
+  if [[ ! -d $(find ${runtimeFolder} -maxdepth 1 -mindepth 1 -type d -name "groovys*") && ! -f ${forbiddenFile} ]]; then
+    (cd ${runtimeFolder} &&
+        wget -S $(curl -s https://kobo.github.io/groovyserv/download.html \
+        | grep groovyserv \
+        | grep bin.zip \
+        | sed 's/href=/\n/g' \
+        | sed 's/>/\n/g' \
+        | sed 's/</\n/g' \
+        | sed 's/"//g'  \
+        | grep .zip)) &> /dev/null
 
     if [[ $? -ne 0 ]]; then
-      touch ${forbiddenFile}
+        echo "Error downloading groovyserv. Skipping groovyserv setup. Remove ${forbiddenFile} before trying again." > /dev/stderr
+        touch ${forbiddenFile}
     else
-      (cd ${runtimeFolder} && unzip groovyser*.zip && rm groovyser*.zip*)
-      [[ $? -ne 0 ]] && touch ${forbiddenFile}
-      (${runtimeFolder}/groovys*/bin/groovyclient 2&> /dev/null)
-      [[ $? -ne 0 ]] && touch ${forbiddenFile}
+      (cd ${runtimeFolder} && unzip groovyser*.zip && rm groovyser*.zip*) &> /dev/null
+      if [[ $? -ne 0 ]]; then
+        echo "Error unzipping groovyserv. Skipping groovyserv setup. Remove ${forbiddenFile} before trying again." > /dev/stderr
+        touch ${forbiddenFile}
+      fi
+      ${runtimeFolder}/groovys*/bin/groovyclient &> /dev/null
+      if [[ $? -ne 0 ]]; then
+        echo "Error starting groovyserv. Skipping groovyserv setup. Remove ${forbiddenFile} before trying again." > /dev/stderr
+        touch ${forbiddenFile}
+      fi
     fi
   fi
 
-  [[ ! -f ${forbiddenFile} ]] && callerBinary=$(readlink -f $( ls ${runtimeFolder}/groovyserv*/bin/groovyclient 2> /dev/null ) 2> /dev/null) || touch ${forbiddenFile}
+  [[ ! -f ${forbiddenFile} ]] && callerBinary=$(readlink -f $(ls ${runtimeFolder}/groovyserv*/bin/groovyclient) 2> /dev/null) || touch ${forbiddenFile}
   echo $callerBinary
 }
 
 function getExistingOrNewGroovyServPort() {
   # Check first, if GroovyServ is running for the local user!
-  local runningGServInstance=$(ps -ef | grep GroovyServ | grep -v grep | grep `whoami`)
+  local runningGServInstance=$(pgrep -f -a -u $(whoami) GroovyServ)
```

Comment:
Hm, good question. Still, the old version did not work on tbi-pbs4. Whatever we put here, we should check it there.